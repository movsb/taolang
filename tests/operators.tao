function exponentiation() {
    let a;
    
    a = 3;
    a = a ** 2;
    println(a);

    a = 3;
    a **= 2;
    println(a);
}

function multiplication() {
    let a;
    let b;
    let c;

    a = 3;
    a = a * 3;
    println(a);
    a = 3;
    a *= 3;
    println(a);

    b = 8;
    b = b / 2;
    println(b); 
    b = 8;
    b /= 2;
    println(b); 

    c = 8;
    c = c % 3;
    println(c); 
    c = 8;
    c %= 3;
    println(c); 
}

function addition() {
    let a;
    let b;

    a = 3;
    a = a + 3;
    println(a);
    a = 3;
    a += 3;
    println(a);

    b = 8;
    b = b - 2;
    println(b); 
    b = 8;
    b -= 2;
    println(b); 
}

function shift() {
    let a = 255;
    a = a >> 4;
    println(a);
    a <<= 4;
    println(a);

    let b = 15;
    b = b << 4;
    println(a);
    b >>= 4;
    println(b);
}

function comparison() {
    let a = 8;
    let b = 6;

    println(a < b);
    println(b < b);

    println(a <= b);
    println(b <= b);

    println(a > b);
    println(b > b);

    println(a >= b);
    println(b >= b);
}

function equality() {
    let a = 8;
    let b = 6;

    println(a == a);
    println(a == b);
    println(a != b);
}

function bitwise() {
    let a = 15;
    let b = 240;
    let c;

    println(a & b);
    println(a | b);
    println(a ^ b);

    c = a;
    c &= b;
    println(c);

    c = a;
    c |= b;
    println(c);

    c = a;
    c ^= b;
    println(c);

    println(255 &^ 1);
  
    {
        let a = 15;
        a &^= 4;
        println(a);
    }
}

function ternary() {
    println(0 ? 2 : 3);
    println(1 ? 2 : 3);
}

function all() {
    exponentiation();
    multiplication();
    addition();
    shift();
    comparison();
    equality();
    bitwise();
    ternary();
}

function main() {
    all();
}
