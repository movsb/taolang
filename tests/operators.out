9
9
9
9
4
4
2
2
6
6
6
6
15
240
240
15
false
false
false
true
true
false
true
true
true
false
true
0
255
255
0
255
255
254
11
3
2
